jobs/statistics/ProcessUsageStatsLogFile.php

```diff
@@ -0,0 +1,256 @@
+<?php
+
+/**
+ * @file jobs/statistics/ProcessUsageStatsLogFile.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ProcessUsageStatsLogFile
+ *
+ * @ingroup jobs
+ *
+ * @brief Compile context metrics.
+ */
+
+namespace APP\jobs\statistics;
+
+use APP\core\Application;
+use APP\statistics\StatisticsHelper;
+use APP\statistics\TemporaryItemInvestigationsDAO;
+use APP\statistics\TemporaryItemRequestsDAO;
+use APP\statistics\TemporaryTitleInvestigationsDAO;
+use APP\statistics\TemporaryTitleRequestsDAO;
+use APP\statistics\TemporaryTotalsDAO;
+use DateTime;
+use Exception;
+use PKP\core\Core;
+use PKP\db\DAORegistry;
+use PKP\job\exceptions\JobException;
+use PKP\jobs\BaseJob;
+use PKP\statistics\TemporaryInstitutionsDAO;
+use PKP\task\FileLoader;
+
+class ProcessUsageStatsLogFile extends BaseJob
+{
+    /**
+     * The number of times the job may be attempted.
+     */
+    public $tries = 1;
+
+    /**
+     * The load ID = usage stats log file name
+     */
+    protected string $loadId;
+
+    /**
+     * Create a new job instance.
+     */
+    public function __construct(string $loadId)
+    {
+        parent::__construct();
+        $this->loadId = $loadId;
+    }
+
+    /**
+     * Execute the job.
+     */
+    public function handle(): void
+    {
+        $filename = $this->loadId;
+        $dispatchFilePath = StatisticsHelper::getUsageStatsDirPath() . '/' . FileLoader::FILE_LOADER_PATH_DISPATCH . '/' . $filename;
+        if (!file_exists($dispatchFilePath)) {
+            throw new JobException(__(
+                'admin.job.processLogFile.fileNotFound',
+                ['file' => $dispatchFilePath]
+            ));
+        }
+        $this->process($dispatchFilePath);
+    }
+
+
+    protected function process(string $dispatchFilePath): void
+    {
+        $fhandle = fopen($dispatchFilePath, 'r');
```

Comment:
I see some similar code regarding the PR at the OJS, perhaps this could be turned into an abstract/base job class.